### PR TITLE
restrict response format for faa applications_controller actions

### PR DESCRIPTION
**Pull request by vkghub**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [ ] For all UI changes, there is cucumber coverage
- [ ] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [ ] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [ ] There are no inline styles added
- [ ] There are no inline javascript added
- [ ] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [ ] Code does not use .html_safe
- [ ] All images added/updated have alt text
- [x] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/187465763

# A brief description of the changes

Current behavior: faa applications_controller actions responds to all MIME types

New behavior: restrict response formats for faa applications_controller actions, Remove unused end points.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

File: components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb [hunk 1]
```diff
@@ -46,28 +48,30 @@ def index_with_filter
...
+        end
       else
-        render json: result.failure.to_h, status: 422
+        respond_to do |format|
+          format.json { render json: result.failure.to_h, status: 422 }
+        end
       end
     end
 
-    def new
```
Comment (ymhari) [inline components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb:54]: Why did we take out this?

File: components/financial_assistance/app/views/financial_assistance/applications/index_with_filter.html.erb [hunk 2]
```diff
@@ -60,7 +60,7 @@
                   ] %>
                 <% if current_user.has_hbx_staff_role? %>
                   <% dropdown << ['Full Application', raw_application_application_path(application), application.is_reviewable? ? 'static' : 'disabled'] %>
-                  <% dropdown << ['Transfer History', transfer_history_application_path(application), 'static' ] if FinancialAssistanceRegistry.feature_enabled?(:transfer_history_page) %>
+                  <% dropdown << ['Transfer History', transfer_history_application_path(application), 'static' ] if true %>
```
Comment (ymhari) [inline components/financial_assistance/app/views/financial_assistance/applications/index_with_filter.html.erb:63]: This should be a push by mistake.

File: components/financial_assistance/config/routes.rb [hunk 3]
```diff
@@ -12,7 +12,6 @@
     put ':step/:step', on: :member, action: 'step'
     post :step, on: :collection
     get 'step/:step', on: :member, action: 'step', as: 'go_to_step'
-    get 'help_paying_coverage', on: :collection, action: 'help_paying_coverage', as: 'help_paying_coverage'
```
Comment (ymhari) [inline components/financial_assistance/config/routes.rb:15]: Taking out the route? Not related to this ticket.

File: components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb [hunk 4]
```diff
@@ -46,28 +48,30 @@ def index_with_filter
...
-
-
       load_support_texts
+
+      respond_to :html
     end
 
     # rubocop:disable Metrics/AbcSize
     def step
+      raise ActionController::UnknownFormat unless request.format.html?
```
Comment (ymhari) [inline components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb:73]: Good one here.

**Review by ymhari — CHANGES_REQUESTED** [review]

File: components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb (hunk 1 above)
Comment (vkghub) [reply components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb:54]: we are not using these end points in FAA applications controller
- new
- help paying for coverage

File: components/financial_assistance/app/views/financial_assistance/applications/index_with_filter.html.erb (hunk 2 above)
Comment (vkghub) [reply components/financial_assistance/app/views/financial_assistance/applications/index_with_filter.html.erb:63]: good catch, I will remove

File: components/financial_assistance/config/routes.rb (hunk 3 above)
Comment (vkghub) [reply components/financial_assistance/config/routes.rb:15]: It many not be related to this ticket, but if we keep this, we have to add mime type for this action and cannot be testable. It is better to clean up.

**Review by ymhari — APPROVED** [review]

**Review by TristanB17 — APPROVED** [review]

**Pull request merged**